gadget/install/install.go

```diff
@@ -160,14 +175,32 @@ func Run(gadgetRoot, device string, options Options, observer SystemInstallObser
 		}
 	}
 
+	dataKeySet := keysForRoles[gadget.SystemData]
+	if dataKeySet == nil {
+		return fmt.Errorf("internal error: system data encryption key set is unset")
+	}
+
 	// Write the recovery key
 	recoveryKeyFile := filepath.Join(boot.InstallHostFDEDataDir, "recovery.key")
-	if err := rkey.Save(recoveryKeyFile); err != nil {
+	if err := dataKeySet.rkey.Save(recoveryKeyFile); err != nil {
 		return fmt.Errorf("cannot store recovery key: %v", err)
 	}
 
 	if observer != nil {
-		observer.ChosenEncryptionKey(key)
+		observer.ChosenEncryptionKey(dataKeySet.key)
+	}
+
+	saveKeySet := keysForRoles[gadget.SystemSave]
+	if saveKeySet != nil {
```

Comment:
to unblock us is ok, but I think we should make saving keys a problem of the caller and not of gadget/install anymore

Reply:
I can propose a separate PR.

Reply:
Yeah, let's make sure this is not forgotten, ideally we had a "TODO:UC20:" here

Reply:
The PR is proposed here: #9476